### PR TITLE
Form Recognizer builder clients

### PR DESCRIPTION
**Pull request by samvaity**
- Remove `getTrainingClient()` from FormRecognizer clients.
- Add `getFormRecognizerAsyncClient()` to FormTraining clients.
- Add FormTrainingClientBuilder to independently build sync and async training clients.

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,8 @@
 # Release History
 
 ## 1.0.0-beta.3 (Unreleased)
-
+- Add `getFormRecognizerClient()` and `getFormRecognizerAsyncClient()` in FormTrainingAsyncClient and FormTrainingClient
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md:4]: Nit: The order feels off.

- Add `getFormRecognizerClient()` and `getFormRecognizerAsyncClient()` in FormTrainingClient and FormTrainingAsyncClient

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingAsyncClient.java [hunk 2]
```diff
@@ -58,7 +59,7 @@
  * @see FormRecognizerClientBuilder
```
Comment (srnagar) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingAsyncClient.java:59]: Fix javadoc 
```suggestion
 * @see FormTrainingClientBuilder
```

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingClient.java [hunk 3]
```diff
@@ -33,7 +33,7 @@
  * @see FormRecognizerClientBuilder
```
Comment (srnagar) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingClient.java:33]: Same here - fix the javadoc.

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingAsyncClient.java [hunk 4]
```diff
@@ -58,7 +59,7 @@
...
  * @see FormRecognizerAsyncClient
  */
-@ServiceClient(builder = FormRecognizerClientBuilder.class, isAsync = true)
+@ServiceClient(builder = FormTrainingClientBuilder.class, isAsync = true)
 public class FormTrainingAsyncClient {
```
Comment (srnagar) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingAsyncClient.java:63]: Make both sync and async clients `final`.

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingClientBuilder.java [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * This class provides a fluent builder API to help instantiation of {@link FormTrainingClient FormRecognizerClients}
```
Comment (srnagar) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingClientBuilder.java:40]:
```suggestion
 * This class provides a fluent builder API to help instantiation of {@link FormTrainingClient FormTrainingClients}
```

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingClientBuilder.java [hunk 6]
```diff
@@ -0,0 +1,345 @@
...
+ * </p>
+ *
+ * <p>
+ * Another way to construct the client is using a {@link HttpPipeline}. The pipeline gives the client an
+ * authenticated way to communicate with the service. Set the pipeline with {@link #pipeline(HttpPipeline) this} and
+ * set the service endpoint with {@link #endpoint(String) this}. Using a
+ * pipeline requires additional setup but allows for finer control on how the {@link FormTrainingClient} and
+ * {@link FormTrainingAsyncClient} is built.
+ * </p>
+ *
```
Comment (srnagar) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingClientBuilder.java:57]: Add codesnippets to javadoc.

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingClient.java [hunk 7]
```diff
@@ -44,11 +44,21 @@
...
         this.client = formTrainingAsyncClient;
     }
 
+    /**
+     * Creates a new {@link FormRecognizerClient} object. The new {@code FormTrainingClient}
+     * uses the same request policy pipeline as the {@code FormTrainingClient}.
+     *
+     * @return A new {@code FormRecognizerClient} object.
+     */
+    public FormRecognizerClient getFormRecognizerClient() {
```
Comment (srnagar) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingClient.java:57]: Is this API necessary? The user can just build using the `FormRecognizerClientBuilder` directly.

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingClientBuilder.java [hunk 8]
```diff
@@ -0,0 +1,345 @@
...
+    /**
+     * Sets the {@link FormRecognizerServiceVersion} that is used when making API requests.
+     * <p>
+     * If a service version is not provided, the service version that will be used will be the latest known service
+     * version based on the version of the client library being used. If no service version is specified, updating to a
+     * newer version the client library will have the result of potentially moving to a newer service version.
+     *
+     * @param version {@link FormRecognizerServiceVersion} of the service to be used when making requests.
+     *
+     * @return The updated FormRecognizerClientBuilder object.
```
Comment (srnagar) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingClientBuilder.java:339]: Fix this in all the methods of the builder.
```suggestion
     * @return The updated FormTrainingClientBuilder object.
```

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingClient.java (hunk 7 above)
Comment (samvaity) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingClient.java:57]: Yeah, we are trying to see the discoverability/usability of the FormRecognizer client via the training client. Moreso, don't want users to use another builder to but just the get the `FormRecognizerClient` using the current builder configurations.

**Pull request merged**